skopt/tests/test_space.py

```diff
@@ -375,6 +377,27 @@ def test_categorical_identity():
     assert_array_equal(transformed, samples)
     assert_array_equal(samples, cat.inverse_transform(transformed))
 
+@pytest.mark.fast_test
+def test_categorical_repr():
+
+    # ensure printing of Categorical objects does not
+    # raise an exception. Hhowever we do not insist
+    # on the exact output format.
+
+    import cStringIO as StringIO
+
+    for categories in (
+        range(3), # short ( < 7 items) list of categories
+        range(8), # long ( >= 7 items) list of categories with ellipsis
+        ):
+
+        categories = list(categories)
+        cat = Categorical(categories, transform="identity")
+
+        out = StringIO.StringIO()
+
+        print(cat, file = out)
```

Comment:
Instead of using this hand made solution check out https://docs.pytest.org/en/3.2.3/capture.html#accessing-captured-output-from-a-test-function for capturing stdout and then `assert`ing its contents. This should also fix the travis failures.

Reply:
Thanks for pointing this out (I would not have thought that such a solution existed :-) ), a406a8a has this now . It does not check the actual output format but essentially only assures that no exception is thrown. 

Do you want the test to also to require the exact output string it produces at the moment ?